### PR TITLE
MOSIP-32461 Java21 to develop

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: registration-processor/pre-processor/registration-processor-quality-classifier-stage/src/main/java/io/mosip/registration/processor/quality/classifier/stage/QualityClassifierStage.java [hunk 1]
```diff
@@ -1,492 +1,493 @@
...
+
+		HashMap<String, Float> bioTypeMinScoreMap = new HashMap<String, Float>();
+
+		// get individual biometrics file name from id.json
+		for (BIR bir : birs) {
+
+			if (bir.getOthers() != null) {
+				HashMap<String, String> othersInfo = bir.getOthers();
+				boolean exceptionValue = false;
+				if(othersInfo!=null) {
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-quality-classifier-stage/src/main/java/io/mosip/registration/processor/quality/classifier/stage/QualityClassifierStage.java:428]: why are we doing this change ? Is this associated to any JIRA bug id ?

File: registration-processor/registration-processor-core/pom.xml [hunk 2]
```diff
@@ -1,210 +1,208 @@
...
+			<groupId>io.vertx</groupId>
+			<artifactId>vertx-core</artifactId>
+			<version>${vertx.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>io.vertx</groupId>
+			<artifactId>vertx-config</artifactId>
+		</dependency>
+		<dependency>
+           <groupId>io.zipkin.zipkin2</groupId>
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/pom.xml:183]: what is the use of this library ?

File: registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/Utilities.java [hunk 3]
```diff
@@ -703,35 +612,7 @@ public JSONObject idrepoRetrieveIdentityByRid(String regId) throws ApisResourceA
...
-					"", "Utilities::calculateAge():: error with error message "
-							+ PlatformErrorMessages.RPR_SYS_PARSING_DATE_EXCEPTION.getMessage());
-			throw new ParsingException(PlatformErrorMessages.RPR_SYS_PARSING_DATE_EXCEPTION.getCode(), e);
-		}
-		LocalDate ld = new java.sql.Date(birthDate.getTime()).toLocalDate();
-		Period p = Period.between(ld, LocalDate.now());
-		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(), "",
-				"Utilities::calculateAge():: exit");
 
-		return p.getYears();
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/Utilities.java:732]: These changes are again related to any bugfix ?

Comment (vishwa-vyom) [reply registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/Utilities.java:732]: why are these methods moved to the new file ?

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 4]
```diff
@@ -1,622 +1,627 @@
...
+		String registrationType = workflowInternalActionDTO.getReg_type();
+
+		regProcLogger.info("processAnonymousProfile called for registration id {}", registrationId);
+
+		InternalRegistrationStatusDto registrationStatusDto = registrationStatusService.getRegistrationStatus(
+				registrationId, registrationType, workflowInternalActionDTO.getIteration(),
+				workflowInternalActionDTO.getWorkflowInstanceId());
+		JSONObject regProcessorIdentityJson = utility.getRegistrationProcessorMappingJson(MappingJsonConstants.IDENTITY);
+		String idSchemaVersionValue = JsonUtil.getJSONValue(JsonUtil.getJSONObject(regProcessorIdentityJson, MappingJsonConstants.IDSCHEMA_VERSION), MappingJsonConstants.VALUE);
+		String schemaVersion = priorityBasedpacketManagerService.getFieldByMappingJsonKey(registrationId,
```
Comment (vishwa-vyom) [inline registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java:274]: why is this change required ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]
Did a quick high level review and added comments.

File: registration-processor/pre-processor/registration-processor-quality-classifier-stage/src/main/java/io/mosip/registration/processor/quality/classifier/stage/QualityClassifierStage.java (hunk 1 above)
Comment (sowmya695) [reply registration-processor/pre-processor/registration-processor-quality-classifier-stage/src/main/java/io/mosip/registration/processor/quality/classifier/stage/QualityClassifierStage.java:428]: Jana added some changes in bio-utils,because of that it was not working in registration processor.Then we did some changes there I have added null check and some code changes.I reverted those,Null check i have not removed.Jana reverted his changes.

File: registration-processor/registration-processor-core/pom.xml (hunk 2 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/pom.xml:183]: I am getting error while running vertx application after migration  "zipkin2.Endpoint$Builder cannot be resolved. It is indirectly referenced from required type brave.http.HttpServerAdapter" Thats why added

File: registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/Utilities.java (hunk 3 above)
Comment (sowmya695) [reply registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/Utilities.java:732]: These changes are related to circular dependency which I got in registration processor after migration while running application.So as suggested by Loga I have break the dependency  and moved code to utility

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java (hunk 4 above)
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java:274]: To resolve circular depdency moved code to utility and resolved issues.

**Review by vishwa-vyom — APPROVED** [review]
